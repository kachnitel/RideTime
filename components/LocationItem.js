import React from 'react';
import { Text, View, StyleSheet } from 'react-native';
import { LocationItemDetail } from './LocationItemDetail';

export class LocationItem extends React.Component {
  render() {
    return <View style={styles.location}>
      <Text style={styles.name}>{this.props.location.name}</Text>
      <LocationItemDetail location={this.props.location} />
      {/* <Text style={styles.itemCoords}>{item.coords[0]}; {item.coords[1]}</Text> */}
    </View>;
  }
}

const styles = StyleSheet.create({
    name: {
      padding: 3,
      fontSize: 16,
      height: 26,
    },
    itemCoords: {
      paddingLeft: 5,
      fontSize: 10,
      height: 12,
    },
    location: {
      height: 64,
      backgroundColor: 'white',
      borderRadius: 4,
      borderWidth: 0.5,
      borderColor: '#d6d7da',
    }
})