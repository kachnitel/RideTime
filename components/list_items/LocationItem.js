import React from 'react'
import { StyleSheet, Text, View } from 'react-native'
import Layout from '../../constants/Layout'
import LocationItemDetail from '../LocationItemDetail'
import PropTypes from 'prop-types'

export default class LocationItem extends React.Component {
  render () {
    return (
      <View style={{ ...styles.listItem, ...this.props.style }}>
        <Text style={{ ...styles.name, ...this.props.style }} numberOfLines={1} >
          {this.props.location.name}
        </Text>
        <LocationItemDetail location={this.props.location} />
      </View>
    )
  }
}

LocationItem.propTypes = {
  location: PropTypes.object,
  style: PropTypes.any
}

const styles = StyleSheet.create({
  name: {
    fontSize: Layout.window.hp(2.75),
    fontWeight: 'bold',
    flex: 1
  },
  listItem: {
    height: Layout.window.hp(15),
    paddingVertical: Layout.window.hp(1.5),
    paddingHorizontal: Layout.window.wp(4)
  }
})
