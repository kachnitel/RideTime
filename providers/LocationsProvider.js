// TODO: filter by visible frame https://github.com/reggie3/react-native-webview-leaflet/blob/8e5830fc23d121db19f51d7dea872d553c253ba5/web/mapComponent.js#L307
import { get } from '../src/Connection'

export default class LocationsProvider {
  /*
     * TODO: Define ranges of lat/lon if map provides corners
     * otherwise let server figure it out
     * @param latMin
     * @param latMax
     * @param lonMin
     * @param lonMax
     *
     */
  getLocations = () => {
    return get('api/locations')
  }
}
