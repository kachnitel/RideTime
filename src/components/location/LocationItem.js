import React from 'react'
import PropTypes from 'prop-types'
import { inject, observer } from 'mobx-react/native'
import {
  StyleSheet,
  Text,
  View,
  ActivityIndicator
} from 'react-native'
import Layout from '../../../constants/Layout'
import LocationItemDetail from './LocationItemDetail'
import { Location } from '../../stores/LocationStore.mobx'
import TrailforksLink from '../TrailforksLink'

export default
@inject('LocationStore')
@observer
class LocationItem extends React.Component {
  location: Location
  state = {
    loading: true
  }

  async componentDidMount () {
    this.location = await this.props.LocationStore.getAsync(this.props.locationId)
    this.setState({ loading: false })
  }

  render () {
    return (
      this.state.loading
        ? <ActivityIndicator />
        : <View style={styles.container}>
          <View style={styles.headerContainer}>
            <Text style={styles.name} numberOfLines={1} >
              {this.location.name}
            </Text>
            <TrailforksLink relativeUrl={'region/' + this.location.alias + '/'} />
          </View>
          <LocationItemDetail location={this.location} />
        </View>
    )
  }
}

LocationItem.propTypes = {
  locationId: PropTypes.number,
  style: PropTypes.any
}

const styles = StyleSheet.create({
  name: {
    fontSize: Layout.window.hp(2.75),
    flex: 1
  },
  headerContainer: {
    flexDirection: 'row'
  }
})
