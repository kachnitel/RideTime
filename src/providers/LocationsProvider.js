import ApiConnection from '../ApiConnection'

export default class LocationsProvider {
  get = (id) => {
    return ApiConnection.get('api/locations/' + id)
  }

  filter = (filter: Object, params: Object) => {
    return ApiConnection.get('api/locations', {
      filter: filter,
      ...params
    })
  }

  trailsFilter = (filter: Object) => {
    return ApiConnection.get('api/locations/trails', {
      filter: {
        activitytype: 1,
        ...filter
      }
    })
  }

  routesFilter = (filter: Object) => {
    return ApiConnection.get('api/locations/routes', {
      filter: {
        activitytype: 1,
        ...filter
      }
    })
  }
}
