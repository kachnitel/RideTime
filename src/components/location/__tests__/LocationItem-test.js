/* eslint-env jest */
import React from 'react'
import renderer from 'react-test-renderer'
import LocationItem from '../LocationItem'
import LocationStore, { Location } from '../../../stores/LocationStore.mobx'
import LocationsProvider from '../../../providers/LocationsProvider'

jest.mock('../LocationDifficulties', () => 'LocationDifficulties')
jest.mock('../../ride/RiderCount', () => 'RiderCount')

it('renders correctly', async () => {
  let testLocation = {
    id: 1,
    name: 'Place',
    difficulties: [1, 2]
  }

  let locationStore = new LocationStore(new LocationsProvider(), {})

  let location = new Location(locationStore)
  location.populateFromApiResponse(testLocation)
  locationStore.add(location)

  const component = renderer.create(
    <LocationItem.wrappedComponent
      locationId={location.id}
      LocationStore={locationStore}
    />
  )
  expect(component.toJSON()).toMatchSnapshot()
  let instance = component.getInstance()
  await new Promise(resolve => process.nextTick(resolve))
  expect(instance.location).toBe(location)
  expect(component.toJSON()).toMatchSnapshot()
})
