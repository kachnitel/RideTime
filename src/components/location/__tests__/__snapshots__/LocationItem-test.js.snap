// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<ActivityIndicator
  animating={true}
  color="#999999"
  hidesWhenStopped={true}
  size="small"
  style={
    Object {
      "height": 133.5,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
/>
`;

exports[`renders correctly 2`] = `
<View
  style={
    Object {
      "height": 133.5,
      "paddingHorizontal": 30,
      "paddingVertical": 20,
    }
  }
>
  <View
    style={
      Object {
        "flexDirection": "row",
      }
    }
  >
    <Text
      numberOfLines={1}
      style={
        Object {
          "flex": 1,
          "fontSize": 36.5,
        }
      }
    >
      Place
    </Text>
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "backgroundColor": "#ffbcbc",
            "borderColor": "red",
            "borderWidth": 1,
            "height": 100,
            "justifyContent": "center",
            "margin": 10,
            "width": 300,
          },
          undefined,
        ]
      }
    >
      <Text
        style={
          Object {
            "color": "#333333",
            "margin": 20,
          }
        }
      >
        TouchableNativeFeedback is not supported on this platform!
      </Text>
    </View>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "space-between",
      }
    }
  >
    <LocationDifficulties
      difficulties={
        Array [
          1,
          2,
        ]
      }
      iconSize={53.5}
      style={
        Object {
          "backgroundColor": "rgba(220, 220, 220, .5);",
          "borderRadius": 11.5,
          "flexDirection": "row",
          "padding": 4,
        }
      }
    />
  </View>
</View>
`;
